.github/workflows/sync-submodules-and-deps.yaml

```diff
@@ -52,3 +52,26 @@ jobs:
           token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
It is indeed adding operational cost.

Since we currently don't have a mechanism for building/distributing any particular version of the nerdctl-full archive which is how Lima installs nerdct/containerd/etc., checking these versions at runtime is the only way to be sure we're using the expected version.

Instead of letting Lima updates potentially update these versions without our knowledge, we now will have to consciously update the version in order for the tests to pass.

Relying on Lima's default Lima nerdctl-full archive is not a good or bad thing, it's just the way we currently operate. Unless we deviate from this model to e.g. always build a custom nerdctl-full in finch-core, this is the only way we can gate which version of these dependencies is being used is at runtime.

An example of where this would have been useful is when we did deviate from Lima's default nerdctl-full archive ([PR](https://github.com/runfinch/finch/pull/783)) for the [v1.1.0 release](https://github.com/runfinch/finch/releases/tag/v1.1.0) due to the security issues mentioned in the release notes, but we did so only partially, which causes the much more operationally expensive operation of recalling the Windows installer and re-releasing v1.1.1. One reason we were unaware of this partial update to the versions of runc/buildkit on Windows was because we don't explicitly validate their versions in this test.